packages/block-library/src/post-title/edit.js

```diff
@@ -83,15 +83,19 @@ export default function PostTitleEdit( {
 
 	if ( isLink ) {
 		titleElement = (
-			<a
-				href={ link }
-				target={ linkTarget }
-				rel={ rel }
-				onClick={ ( event ) => event.preventDefault() }
-				{ ...blockProps }
-			>
-				{ titleElement }
-			</a>
+			<TagName { ...blockProps } >
+				<PlainText
+					tagName="a"
+					placeholder={ title.length === 0 ? __( 'No Title' ) : null }
```

Comment:
I needed to add this check to make sure that the placeholder was not read by the screen reader when there was a title.